core/src/main/java/org/opensearch/sql/expression/ip/IPFunction.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.expression.ip;
+
+import static org.opensearch.sql.data.type.ExprCoreType.BOOLEAN;
+import static org.opensearch.sql.data.type.ExprCoreType.STRING;
+import static org.opensearch.sql.expression.function.FunctionDSL.define;
+import static org.opensearch.sql.expression.function.FunctionDSL.impl;
+import static org.opensearch.sql.expression.function.FunctionDSL.nullMissingHandling;
+
+import inet.ipaddr.AddressStringException;
+import inet.ipaddr.IPAddressString;
+import inet.ipaddr.IPAddressStringParameters;
+import lombok.experimental.UtilityClass;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.data.model.ExprValueUtils;
+import org.opensearch.sql.exception.SemanticCheckException;
+import org.opensearch.sql.expression.function.BuiltinFunctionName;
+import org.opensearch.sql.expression.function.BuiltinFunctionRepository;
+import org.opensearch.sql.expression.function.DefaultFunctionResolver;
+
+/** Utility class that defines and registers IP functions. */
+@UtilityClass
+public class IPFunction {
```

Comment:
My _intent_ here was to align this name with the other similar "function" utility classes, of which the vast majority are not pluralized. That being said, I agree that the pluralized form is a better name here, so I have renamed _all_ the relevant classes to be plural (including the associated unit and integration classes):
- `AggregatorFunctions`
- `DateTimeFunctions`
- `IPFunctions`
- `ArithmeticFunctions`
- `MathematicalFunctions`
- `TypeCastOperators` (ignoring whether these are actually "operators" or "functions"...)
- `BinaryPredicateOperators`
- `UnaryPredicateOperators` (again, ignoring whether these are really "functions" rather than "operators")
- `SystemFunctions` (not renamed - was already plural)
- `TextFunctions`

Hope that addresses your comment (and also generally improves naming).